rbx_types/src/terrain.rs

```diff
@@ -0,0 +1,548 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    convert::TryFrom,
+    fmt,
+};
+
+use crate::Vector3;
+
+/// Size of a chunk. Chunks are cubes, so this is the length/width/height.
+const CHUNK_SIZE: i32 = 2i32.pow(5);
+
+/// Coordinates of a chunk or a voxel. For internal use.
+// Can't use Vector3int16; we need a 32 bit integer.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct TerrainVec {
+    pub x: i32,
+    pub y: i32,
+    pub z: i32,
+}
+
+impl TerrainVec {
+    pub fn from_vec3(i: Vector3) -> Self {
+        Self {
+            x: i.x as i32,
+            y: i.y as i32,
+            z: i.z as i32,
```

Comment:
This isn't really all that rigorous. I don't know if we should expose this without actually checking if the `f32` is a valid `i32`. At the same time though, that's a hard thing to check, so maybe it's fine.

We should for sure document how it converts it though. Something like "naively converts the components to be `i32`s" would at least make it not surprising.

Reply:
the truncation is actually desirable behavior imo, supposed to be whole number and rounding down makes the most sense. or could remove this conversion entirely and force you to know exactly what you're doing